config_examples/masking_logbody.yaml

```diff
@@ -0,0 +1,36 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+exporters:
+  otlphttp:
+    endpoint: ${env:DT_ENDPOINT}
+    headers:
+      Authorization: "Api-Token ${env:API_TOKEN}"
+
+processors:
+  transform/cc:
+    error_mode: ignore
+    trace_statements:
+    log_statements:
+      - context: log
+        statements:
+          - replace_pattern(body, "3\\s*[47](\\s*[0-9]){9}((\\s*[0-9]){4})$", "**** $$2")
```

Comment:
Doesnt the `$` at the end mean that this only identifies numbers if the log message ends with the credit card number? I.e.: 
```
2024/11/08 09:54 [paymentservice] This is a log message that contains this credit card number: 2367 8901 2345 6789
```
Would be redacted but for this it wouldn't work:
```
2024/11/08 09:54 [paymentservice] Customer ABC used AMEX 2367 8901 2345 6789 to pay for transaction 12345.
```

Reply:
yes, that's true, I guess I can adapt it a bit to make the regex more general

Reply:
done